sql/core/src/main/scala/org/apache/spark/sql/execution/debug/package.scala

```diff
@@ -116,6 +168,30 @@ package object debug {
     }
   }
 
+  implicit class DebugStreamQuery(query: StreamingQuery) extends Logging {
+    def debug(): Unit = {
+      query match {
+        case w: StreamExecution =>
```

Comment:
The expectations is that `query.debug` and `query.debugCodegen` should work correct ? I think you need to match `StreamingQuery` and extract the `StreamExecution` here.

At-least in the spark-shell the return value for `DataStreamWriter.start` is a StreamingQueryWrapper.
```
scala> import org.apache.spark.sql.execution.debug._
import org.apache.spark.sql.execution.debug._

scala> val query = windowedCounts.writeStream.outputMode("complete").format("console").option("truncate", "false").start()
query: org.apache.spark.sql.streaming.StreamingQuery = org.apache.spark.sql.execution.streaming.StreamingQueryWrapper@6eb825e4
scala> query.debug
Only supported for StreamExecution.
```

Reply:
My bad. Fixed. Also changed the unit tests so that your reported case can be covered in test.